src/Tasks/Microsoft.NET.Build.Tasks/EmbedAppNameInHost.cs

```diff
@@ -0,0 +1,169 @@
+                throw new BuildErrorException(Strings.FileNameIsTooLong, AppBinaryName);
+            }
+
+            var array = File.ReadAllBytes(AppHostSourcePath);
+            var placeHolder = "c3ab8ff13720e8ad9047dd39466b3c8974e592c2fa383d4a3960714caef0c4f2"; //hash value embedded in apphost.exe
+            var bytesToSearch = Encoding.UTF8.GetBytes(placeHolder);
+
+            SearchAndReplace(array, bytesToSearch, bytesToWrite);
+
+
+            if ( !Directory.Exists(destinationDirectory))
+            {
+                Directory.CreateDirectory(destinationDirectory);
+            }
+            
+            File.WriteAllBytes(_modifiedAppHostPath, array);
+
+        }
+
+        // See: https://en.wikipedia.org/wiki/Knuth%E2%80%93Morris%E2%80%93Pratt_algorithm
+        static int[] ComputeKMP_FailureFunction(byte[] pattern)
+        {
+            int[] table = new int[pattern.Length];
+            if (pattern.Length >= 1)
+            {
+                table[0] = -1;
+            }
+            if (pattern.Length >= 2)
+            {
+                table[1] = 0;
+            }
+
+            int pos = 2;
+            int cnd = 0;
+            while (pos < pattern.Length)
+            {
+                if (pattern[pos - 1] == pattern[cnd])
+                {
+                    table[pos] = cnd + 1;
+                    cnd++;
+                    pos++;
+                }
+                else if (cnd > 0)
+                {
+                    cnd = table[cnd];
+                }
+                else
+                {
+                    table[pos] = 0;
+                    pos++;
+                }
+            }
+            return table;
+        }
+
+        // See: https://en.wikipedia.org/wiki/Knuth%E2%80%93Morris%E2%80%93Pratt_algorithm
+        static  int KMP_Search(byte[] pattern, byte[] bytes)
+        {
+            int m = 0;
+            int i = 0;
+            int[] table = ComputeKMP_FailureFunction(pattern);
+
+            while (m + i < bytes.Length)
+            {
+                if (pattern[i] == bytes[m + i])
+                {
+                    if (i == pattern.Length - 1)
+                    {
+                        return m;
+                    }
+                    i++;
+                }
+                else
+                {
+                    if (table[i] > -1)
+                    {
+                        m = m + i - table[i];
+                        i = table[i];
+                    }
+                    else
+                    {
+                        m++;
+                        i = 0;
+                    }
+                }
+            }
+
+            return -1;
+        }
+
+        public static  void SearchAndReplace(byte[] array, byte[] search, byte[] replace)
+        {
+
+            int offset = KMP_Search(search, array);
+            if (offset < 0)
+            {
+                throw new BuildErrorException(Strings.AppHostHasBeenModified);
```

Comment:
This string has an arg for it:

`<value>The App host does not have the embedded hash value  '{0}'</value>`, but I don't see you providing it here.